spec/classes/agent_spec.rb

```diff
@@ -63,6 +63,25 @@
           })
         }
 
+        service_env_vars_content = ['# This file is being maintained by Puppet.','# DO NOT EDIT'].join("\n")
```

Comment:
This is a programmatic way to build the content which is ok for the code, but not as great for the spec testing as it might miss something. Prefer doing it in such a way that you could copy/paste from an actual file and read it would be in the generated file. Here is an example.

https://github.com/ghoneycutt/puppet-module-bind/blob/master/spec/classes/init_spec.rb#L33-L78

Reply:
That's harder to read in my opinion. I build the string using array then join with newlines and say `with_content(...)` which matches exact content. It's the same end result as your example but without the complication of handling indented HEREDOC.  Both approaches match exact content, so won't miss anything. Should I still change to the example from bind module?

Reply:
Please do. I prefer the HEREDOC approach for testing so that you can read the document as you would when it is generated. For things larger than ~10 lines or so, I would use a fixture file.

Reply:
Updated to use HEREDOC.